extensions/notebook/src/dialog/managePackages/addNewPackageTab.ts

```diff
@@ -0,0 +1,264 @@
+		await this.packagesSearchButton.updateProperties({ enabled: enable });
+		await this.newPackagesNameLoader.updateProperties({ loading: !enable });
+		await this.newPackagesVersionsLoader.updateProperties({ loading: !enable });
+		await this.newPackagesSummaryLoader.updateProperties({ loading: !enable });
+	}
+
+	private async loadNewPackageInfo(): Promise<void> {
+		await this.packageInstallButton.updateProperties({ enabled: false });
+		await this.toggleNewPackagesFields(false);
+		try {
+			let packageName = this.newPackagesSearchBar.value;
+			if (!packageName || packageName.length === 0) {
+				return;
+			}
+
+			let pipPackage = await this.fetchPypiPackage(packageName);
+			if (!pipPackage.versions || pipPackage.versions.length === 0) {
+				this.dialog.showErrorMessage(this.NoVersionsFoundError);
+				return;
+			}
+
+			await this.newPackagesName.updateProperties({
+				value: packageName
+			});
+			await this.newPackagesVersions.updateProperties({
+				values: pipPackage.versions,
+				value: pipPackage.versions[0],
+			});
+			await this.newPackagesSummary.updateProperties({
+				value: pipPackage.summary
+			});
+
+			// Only re-enable install on success
+			await this.packageInstallButton.updateProperties({ enabled: true });
+		} catch (err) {
+			this.dialog.showErrorMessage(utils.getErrorMessage(err));
+
+			await this.newPackagesName.updateProperties({
+				value: this.InvalidTextPlaceholder
+			});
+			await this.newPackagesVersions.updateProperties({
+				value: this.InvalidTextPlaceholder,
+				values: [this.InvalidTextPlaceholder],
+			});
+			await this.newPackagesSummary.updateProperties({
+				value: this.InvalidTextPlaceholder
+			});
+		} finally {
+			await this.toggleNewPackagesFields(true);
+		}
+	}
+
+	private async fetchPypiPackage(packageName: string): Promise<PipPackageData> {
+		return new Promise<PipPackageData>((resolve, reject) => {
+			request.get(`https://pypi.org/pypi/${packageName}/json`, { timeout: 10000 }, (error, response, body) => {
+				if (error) {
+					return reject(error);
+				}
+
+				if (response.statusCode === 404) {
+					return reject(this.PackageNotFoundError);
+				}
+
+				if (response.statusCode !== 200) {
+					return reject(response.statusMessage);
+				}
+
+				let packagesJson = JSON.parse(body);
+				let versionKeys = Object.keys(packagesJson.releases);
+				versionKeys = versionKeys.filter(versionKey => {
+					let releaseInfo = packagesJson.releases[versionKey];
+					return Array.isArray(releaseInfo) && releaseInfo.length > 0;
+				});
+				versionKeys.sort((first, second) => {
+					// sort in descending order
+					let firstVersion = first.split('.').map(numStr => Number.parseInt(numStr));
+					let secondVersion = second.split('.').map(numStr => Number.parseInt(numStr));
+
+					// If versions have different lengths, then append zeroes to the shorter one
+					if (firstVersion.length > secondVersion.length) {
```

Comment:
While your logic looks great, could we just use the `semver` npm package instead? It has `.gt` and `.lt` comparators built in. We already use it in other places in Azure Data Studio.

Reply:
https://www.npmjs.com/package/semver

Reply:
Looks like semver fails on versions with less than 3 parts. "0.1" is an invalid version, for example.